lib/onelogin/api/client.rb

```diff
@@ -1348,6 +1387,229 @@ def get_saml_assertion_verifying(app_id, device_id, state_token, otp_token=nil,
+
+          data = {
+            'factor_id'=> factor_id.to_i,
+            'display_name'=> display_name,
+            'number'=> number
+          }
+
+          response = HTTParty.post(
+            url,
+            :headers => authorized_headers,
+            body: data.to_json
+          )
+
+          if response.code == 200
+            json_data = JSON.parse(response.body)
+            if json_data and json_data['data']
+              return OneLogin::Api::Models::OTPDevice.new(json_data['data'][0])
+            end
+          else
+            @error = response.code.to_s
+            @error_description = extract_error_message_from_response(response)
+          end
+        rescue Exception => e
+          @error = '500'
+          @error_description = e.message
+        end
+
+        nil
+      end
+
+      # Return a list of authentication factors registered to a particular user for multifactor authentication (MFA)
+      #
+      # @param user_id [Integer] The id of the user.
+      #
+      # @return [Array] OTPDevice List
+      # 
+      # @see {https://developers.onelogin.com/api-docs/1/multi-factor-authentication/enrolled-factors Get Enrolled Authentication Factors documentation}
+      def get_enrolled_factors(user_id)
+        clean_error
+        prepare_token
+
+        begin
+          url = url_for(GET_ENROLLED_FACTORS_URL, user_id)
+
+          response = HTTParty.get(
+            url,
+            :headers => authorized_headers
+          )
+
+          otp_devices = []
+          if response.code == 200
+            json_data = JSON.parse(response.body)
+            if json_data and json_data['data'] and json_data['data']['otp_devices']
+              json_data['data']['otp_devices'].each do |otp_device_data|
+                otp_devices << OneLogin::Api::Models::OTPDevice.new(otp_device_data)
+              end
+            end
+          else
+            @error = response.code.to_s
+            @error_description = extract_error_message_from_response(response)
+          end
+          return otp_devices
+        rescue Exception => e
+          @error = '500'
+          @error_description = e.message
+        end
+
+        nil
+      end
+
+      # Triggers an SMS or Push notification containing a One-Time Password (OTP)
+      # that can be used to authenticate a user with the Verify Factor call.
+      #
+      # @param user_id [Integer] The id of the user.
+      # @param device_id [Integer] The id of the MFA device.
+      #
+      # @return [FactorEnrollmentResponse] Info with User Id, Device Id, and OTP Device
+      # 
+      # @see {https://developers.onelogin.com/api-docs/1/multi-factor-authentication/activate-factor Activate an Authentication Factor documentation}
+      def activate_factor(user_id, device_id)
+        clean_error
+        prepare_token
+
+        begin
+          url = url_for(ACTIVATE_FACTOR_URL, user_id, device_id)
+
+          response = HTTParty.get(
```

Comment:
@pitbulk I was experimenting with your code and I run into 404 error.
After changing `.get` to `.post` it worked smoothly.

Reply:
👍